pkg/kustomize/kustomize.go

```diff
@@ -618,11 +618,19 @@ func Inflate(
 		resources[index] = obj
 	}
 
-	for index, resource := range resources {
-		resources[index] = v1.EnsureOwnerReference(quay, resource)
+	var filteredResources []client.Object
+	for _, resource := range resources {
+		if resource == nil {
+			continue
+		}
+		filteredResources = append(filteredResources, resource)
+	}
+
+	for index, resource := range filteredResources {
```

Comment:
following the suggestion above we could delete the second loop and build our slice in the first loop